### PR TITLE
Partitioner delete md raid

### PR DESCRIPTION
**Pull request by joseivanlopez**
PBI: https://trello.com/c/UddMuRdT/736-2-expert-partitioner-delete-md-raid

### CODE REVIEW

File: src/lib/y2partitioner/actions/delete_md.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      # @see DeleteDevice#dependent_devices
+      # @see DeleteDevice#confirm_recursive_delete
+      # @see DeleteDevice#lvm_vg
+      def confirm_for_used_by_lvm
+        confirm_recursive_delete(
+          dependent_devices,
+          _("Confirm Deleting Md RAID Used by LVM"),
+          # TRANSLATORS: name is the name of the volume group that the md raid
+          #   belongs to (e.g., /dev/system)
+          format(_("The selected md raid is used by volume group \"%{name}\".\n" \
```
Comment (ancorgs) [inline src/lib/y2partitioner/actions/delete_md.rb]: Inconsistent case. What so we usually use in the UI? MD RAID? Md RAID? md raid? None of the above?

File: src/lib/y2partitioner/widgets/pages/md_raids.rb [hunk 2]
```diff
@@ -75,14 +76,26 @@ def contents
...
+        # Table with all md raids
+        #
+        # @return [MdRaidsTable]
+        def table
+          @table ||= MdRaidsTable.new(devices, @pager)
+        end
+
+        # Returns all md raids
+        #
+        # @return [Array<Y2Storage::LvmVg, Y2Storage::LvmLv>]
```
Comment (ancorgs) [inline src/lib/y2partitioner/widgets/pages/md_raids.rb]: Wrong type in yardoc

File: test/y2storage/devicegraph_test.rb [hunk 3]
```diff
@@ -253,4 +253,46 @@ def with_sda2_deleted(initial_graph)
       end
     end
   end
+
+  describe "#remove_md" do
+    subject(:devicegraph) { Y2Storage::StorageManager.instance.staging }
```
Comment (ancorgs) [inline test/y2storage/devicegraph_test.rb]: NP: Maybe a test to see/document what happens if the MD device does not exist?

File: src/lib/y2partitioner/actions/delete_md.rb (hunk 1 above)
Comment (joseivanlopez) [reply src/lib/y2partitioner/actions/delete_md.rb]: Well, when deleting a partition we use RAID. I will use the same here.

File: src/lib/y2partitioner/widgets/pages/md_raids.rb (hunk 2 above)
Comment (joseivanlopez) [reply src/lib/y2partitioner/widgets/pages/md_raids.rb]: Copy-paste

File: test/y2storage/devicegraph_test.rb [hunk 4]
```diff
@@ -253,4 +253,63 @@ def with_sda2_deleted(initial_graph)
...
+
+      expect(descendants_sid).to_not be_empty
+
+      devicegraph.remove_md(md)
+
+      existing_descendants = descendants_sid.map { |sid| devicegraph.find_device(sid) }.compact
+      expect(existing_descendants).to be_empty
+    end
+
+    context "when the md does not exist in the devicegraph" do
```
Comment (ancorgs) [inline test/y2storage/devicegraph_test.rb]: And something that is a little surprising to me, so I would like to see a spec if it's asking too much :wink::

```ruby
it "does not raise any exception"
```

**Pull request merged**